src/Uno.UI/UI/Xaml/Media/Animation/Timeline.cs

```diff
@@ -164,21 +164,23 @@ internal BindingPath PropertyInfo
 		{
 			get
 			{
-				if (_propertyInfo == null)
-				{
-					InitTarget();
-					var target = Target ?? GetTargetFromName();
+				// Don't use the cached _propertyInfo if TargetProperty or Target has the changed.
+				var targetPropertyPath = Storyboard.GetTargetProperty(this);
+				InitTarget();
+				var target = Target ?? GetTargetFromName();
 
+				if (_propertyInfo == null || _propertyInfo.Path != targetPropertyPath)
+				{
+					_propertyInfo?.DataContext = null;
 					_propertyInfo = new BindingPath(
-						path: Storyboard.GetTargetProperty(this),
+						path: targetPropertyPath,
 						fallbackValue: null,
 						precedence: DependencyPropertyValuePrecedences.Animations,
 						allowPrivateMembers: false
 					);
-
-					_propertyInfo.DataContext = target;
 				}
 
+				_propertyInfo.DataContext = target;
```

Comment:
The `DataContext` property is not validating for equality, you'll need to validate it here.